From da61f59bdcab744ba9115ceb8cd27c483e46efcd Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.de>
Date: Thu, 23 May 2013 16:44:15 +0200
Subject: [PATCH] Do not switch webcams in some HP ProBooks to XHCI
Git-Commit: da61f59bdcab744ba9115ceb8cd27c483e46efcd
References: bnc#805804
Patch-Mainline: Never (upstream fix breaks kABI)

Some webcams in some HP ProBooks cannot operate reliably
with XHCI (great package loss). As some other cameras do
work it is unlikely to be a bug in uvcvideo. For now, keep
all cameras at EHCI.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
 drivers/usb/host/pci-quirks.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/host/pci-quirks.c b/drivers/usb/host/pci-quirks.c
index 3c738ec..cff1712 100644
--- a/drivers/usb/host/pci-quirks.c
+++ b/drivers/usb/host/pci-quirks.c
@@ -812,7 +812,7 @@ void usb_enable_xhci_ports(struct pci_dev *xhci_pdev)
 
 	/* this needs to be done in a dirty way to preserve kABI */
 	if (dmi_check_system(bt_excepted_from_switch_table))
-		ports_available &= ~8;
+		ports_available &= ~(64 | 8);
 
 	/* Write XUSB2PR, the xHC USB 2.0 Port Routing Register, to
 	 * switch the USB 2.0 power and data lines over to the xHCI
-- 
1.8.1.4

